Add paragraph about Delivery Overview only available server-side (#5225)
segmentio/segment-docs: src/connections/delivery-overview.md

## How can I configure alerts?
During the Delivery Overview beta, you can use the Event Delivery alerting features (Delivery Alerts) by selecting the **Alerts** tab in the destination header.

## Why is the Delivery Overview page only available for cloud-mode destinations? 

> Similar to Segment's [Event Delivery](https://segment-docs.netlify.app/docs/connections/event-delivery/) feature, the Delivery Overview page is only available for server-side integrations (also known as cloud-mode destinations). You will not be able to use the Delivery Overview page for client side integrations (also known as device-mode destinations) because device-mode data is sent directly to the destination tool's API. In order to report on deliverability, the data must be sent to destinations using a server-side connection. 

Reviewer: ```suggestion
Similar to Segment's [Event Delivery](/docs/connections/event-delivery/) feature, the Delivery Overview page is only available for server-side integrations (also known as cloud-mode destinations). You won't be able to use the Delivery Overview page for client side integrations (also known as device-mode destinations) because device-mode data is sent directly to the destination tool's API. In order to report on deliverability, data must be sent to destinations using a server-side connection. 
```
